src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -1665,16 +1650,87 @@ export class StudyViewPageStore {
             });
             return filterAttributes;
         },
-        default: [],
-        onResult: (result) => {
-            let charts = _.reduce(result, (acc, next) => {
-                acc[getClinicalAttributeUniqueKey(next)] = true;
-                return acc;
-            }, {} as {[uniqueId:string]:boolean});
-            this.changeChartsVisibility(charts);
-        }
+        default: []
+    });
+
+    readonly defaultVisibleAttributesClinicalDataCountData = remoteData<ClinicalDataCountItem[]>({
+        await: () => [this.defaultVisibleAttributes],
+        invoke: async () => {
+            return internalClient.fetchClinicalDataCountsUsingPOST({
+                clinicalDataCountFilter: {
+                    attributes: _.filter(this.defaultVisibleAttributes.result, attr => attr.datatype === 'STRING').map(attr => {
+                        return {
+                            attributeId: attr.clinicalAttributeId,
+                            clinicalDataType: attr.patientAttribute ? 'PATIENT' : 'SAMPLE'
+                        } as ClinicalDataFilter
+                    }),
+                    studyViewFilter: this.defaultFilters
+                } as ClinicalDataCountFilter
+            });
+        },
+        onResult: (data) => {
```

Comment:
lets evaluate whether we really need these onResult handlers, which are kind of an anti-pattern

Reply:
I actually did. Most of unnecessary has been removed or grouped together. The ones left are relying on different endpoints. And I think this is the proper place to update some states.